pkg/build/api/types.go

```diff
@@ -425,6 +429,18 @@ type SourceBuildStrategy struct {
 	ForcePull bool
 }
 
+// JenkinsPipelineStrategy holds parameters specific to a Jenkins Pipeline build.
+// This strategy is experimental.
+type JenkinsPipelineBuildStrategy struct {
+	// JenkinsfilePath is the optional path of the Jenkinsfile that will be used to configure the pipeline
+	// relative to the root of the source repository. If both JenkinsfilePath & Jenkinsfile are
+	// both not specified, this defaults to Jenkinsfile in the root of the specified git repo.
+	JenkinsfilePath string
```

Comment:
JenkinsfilePath should default to being "Jenkinsfile" relative to the
context dir, not relative to the root of the repo (like DcokrefilePath)

On Fri, Apr 15, 2016 at 3:57 PM, Ben Parees <notifications@github.com>
wrote:

> In pkg/build/api/types.go
> <https://github.com/openshift/origin/pull/7949#discussion_r59932409>:
>
> > @@ -425,6 +429,18 @@ type SourceBuildStrategy struct {
> >  	ForcePull bool
> >  }
> >
> > +// JenkinsPipelineStrategy holds parameters specific to a Jenkins Pipeline build.
> > +// This strategy is experimental.
> > +type JenkinsPipelineBuildStrategy struct {
> > +	// JenkinsfilePath is the optional path of the Jenkinsfile that will be used to configure the pipeline
> > +	// relative to the root of the source repository. If both JenkinsfilePath & Jenkinsfile are
> > +	// both not specified, this defaults to Jenkinsfile in the root of the specified git repo.
> > +	JenkinsfilePath string
>
> Should we just reuse contextDir instead of adding this field?
> @smarterclayton <https://github.com/smarterclayton> ?
>
> we already use contextDir to help us locate a Dockerfile in a git repo,
> for example.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/7949/files/15f8dabcad29df00915ad82edd4e0a425894b890#r59932409>
>